src/main/java/com/framgia/controller/admins/AdminController.java

```diff
@@ -0,0 +1,92 @@
+package com.framgia.controller.admins;
+
+import javax.servlet.http.HttpSession;
+
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import com.framgia.bean.UserInfo;
+import com.framgia.controller.BaseController;
+import com.framgia.hepler.ConvertUser;
+
+@Controller
+@RequestMapping(value = "/admin")
+public class AdminController extends BaseController {
+
+	private static final Logger logger = Logger.getLogger(AdminController.class);
+
+	@GetMapping()
+	public String adminHome(HttpSession httpSession, Model model) {
+		logger.info("home page");
+		UserInfo currentUser = (UserInfo) httpSession.getAttribute("currentUser");
+		if(currentUser == null) {
+			//show message first login
+			return "redirect:/";
+		}
+		if(currentUser.getRole().equals("USER")) {
+			//show message : u aren't admin
+			//return home
+			return "redirect:/";
+		}else {	
+			model.addAttribute("user", new UserInfo());
+			model.addAttribute("users", ConvertUser.convertUserToUserInfo(userService.getAll()));
+			return "admin-home";
+		}
+		
+	}
+
+	@RequestMapping(value = "/{id}", method = RequestMethod.GET)
+	public String showUserDetail(@PathVariable("id") Integer id, Model model) {
+		logger.info("detail user");
+		UserInfo userInfo = ConvertUser.convertSingleUserToUserInfo(userService.findById(id));
+		if (userInfo == null) {
+			model.addAttribute("css", "danger");
+			model.addAttribute("msg", "notFound");
+		}
+		model.addAttribute("user", userInfo);
+		return "user";
+	}
+
+	@RequestMapping(value = "/{id}/delete", method = RequestMethod.GET)
```

Comment:
Action này và cả bên dưới trong file này, em viết vào UserController nhé
bắt đầu của các action này là `users` nhé

-- ko clear thì đọc lại REST nhé

Reply:
Vâng chị, lỗi này thì là do em lúc đầu định gom toàn bộ các xử lý cho thằng admin

Reply:
comment trên check lại nhé
em xem cách bố trí cấu trúc project nèk
https://github.com/framgia/ecommerce/tree/develop/src/main/java/com/framgia/controller